sunpy/map/mapbase.py

```diff
@@ -773,8 +773,17 @@ def _validate_meta(self):
             CUNIT1, CUNIT2, WAVEUNIT
 
         """
+        msg = ('Image coordinate units for axis {} not present in metadata. '
+               'It can be set using `map.meta["cunit{}"] = unit`.')
+        err_message = []
+        for i in [1, 2]:
+            if self.meta.get(f'cunit{i}') is None:
+                err_message.append(msg.format(i, i))
 
-        for meta_property in ('cunit1', 'cunit2', 'waveunit'):
+        if err_message:
+            raise AttributeError('\n'.join(err_message))
```

Comment:
As [discussed](https://matrix.to/#/!MeRdFpEonLoCwhoHeT:matrix.org/$15519781816270xRRRX:cadair.com?via=cadair.com&via=matrix.org&via=openastronomy.org) if you make this raise a specific error to map header validation then it will be simple to patch `MapFactory` to skip invalid HDUs.

Reply:
This is then the patch to `map_factory.py` that I think will do the job: https://github.com/Cadair/sunpy/commit/020944b0c56b40b0f63477e7fb1553c12f91d268